pkg/hooks/connection/factory.go

```diff
@@ -133,7 +133,7 @@ func capture(db platform.TestCaseDB, req *http.Request, resp *http.Response, log
 	// err = db.Insert(httpMock, getDeps())
 	err = db.WriteTestcase(&models.TestCase{
 		Version: models.V1Beta2,
-		Name:    "",
+		Name:    pkg.ToYamlHttpHeader(req.Header)["Keploy_test_name"],
```

Comment:
According to good practices the headers shouldn't contain underscores . Convert those underscores to hyphens and also either all the first letters in the header name should be capital or small like `keploy-test-name`. 

Consider both negative and positive cases , what if user doesn't provide `Keploy_test_name` header , will this code `pkg.ToYamlHttpHeader(req.Header)["Keploy_test_name"]` throw any error.

Reply:
in case user doesn't provide keploy_test_name, name field would be empty 
![image](https://github.com/keploy/keploy/assets/91385321/fe91fc57-4d66-42de-9a84-1a32f2661225)

Reply:
got it.